stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -36,49 +35,51 @@ async def create_results_connection(self, query, offset, length, metadata=None):
             if return_obj:
                 return return_obj
 
-            if (result_count == 0 and page_number == 1) or \
-                    (page_number and result_count < self.api_client.result_limit):
+            remaining_record_count = 0
 
-                while result_count < total_records:
-                    query_url = f'{query}&page={page_number}&count={page_size}'
-                    response_wrapper = await self.api_client.get_search_results(query_url, jwt_token)
-                    response_dict, return_obj = self.handle_api_response(response_wrapper)
+            while result_count < total_records:
```

Comment:
There is no reason to keep track of the total number of data handled. The query only requires the page_size and page_number. What is remaining to be handled is decided by the caller in the length option. 

If the user makes a second or (more) request. We don't care how many results were returned in the previous query. We know that the last page was X, so we just continue from there.

Reply:
Removed result_count from meta_data and got it from the pagenumber and pagesize value.
Users can set result_limit from the UI. result_count should not exceed the result_limit.
For this reason, we need the result_count.

Reply:
As discussed in the meeting, length coming from CP4S is always constant (default batch size value- of 2000). Logs from the CP4S will be sent by email. Hence result_count is required to track the total number of records fetched and ensured that it is with in the result_limit. result_count is based on offset.